public/app/features/search/service/frontend.ts

```diff
@@ -0,0 +1,102 @@
+import { DataFrameView, SelectableValue, ArrayVector } from '@grafana/data';
+import { TermCount } from 'app/core/components/TagFilter/TagFilter';
+
+import { DashboardQueryResult, GrafanaSearcher, QueryResponse, SearchQuery } from '.';
+
+export class FrontendSearcher implements GrafanaSearcher {
+  full = new Map<string, FullResultCache>();
+
+  constructor(private parent: GrafanaSearcher) {}
+
+  async search(query: SearchQuery): Promise<QueryResponse> {
+    if (query.facet?.length) {
+      throw new Error('facets not supported!');
+    }
+    // Don't bother... not needed for this exercise
+    if (query.tags?.length || query.ds_uid?.length) {
```

Comment:
or if panel search?

Reply:
I updated this so our cache matches the input request 👍

Reply:
so now it *will* do panel search